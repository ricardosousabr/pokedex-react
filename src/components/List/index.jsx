import { useState } from "react"
import axios from "axios"

export default function List() {
  const [pokemonData, setPokemonData] = useState("")
  const [pokemonName, setPokemonName] = useState("")
  const apiPokemon = "https://pokeapi.co/api/v2/pokemon/" + pokemonName

  function getPokemon() {

    axios.get(apiPokemon)
    .then(response => {
      setPokemonData(response.data)
      console.log(response.data)
    })
    .catch(error => console.log(error))
  }

  function handleSubmit(e) {
    e.preventDefault();

    getPokemon()
  }

  return(
    <div>
      <div>
        <form onSubmit={handleSubmit}>
          <input type="text" onChange={(e) => setPokemonName(e.target.value)}/>
          <button>Search</button>
        </form>
      </div>
      <div>
        <ul>
          {pokemonData ?
            <li>
              <p>{pokemonData.name}</p>
              <img src={pokemonData.sprites.front_default} alt="" width={100} height={100}/>
            </li> : <li>Escreva o nome do seu pokemon</li>
          }
        </ul>
      </div>
    </div>
  )
}