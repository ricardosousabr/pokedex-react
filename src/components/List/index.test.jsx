/**
 * @jest-environment jsdom
 */

import React from 'react'
import {render} from '@testing-library/react'
import List from './index'

const getPokemon = require('./index')

it("renders homepage unchanged", () => {
  const { container } = render(<List />)
  expect(container).toMatchSnapshot()
});
