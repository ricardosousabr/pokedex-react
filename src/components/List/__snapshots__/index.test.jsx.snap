// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div>
    <div>
      <form>
        <input
          type="text"
        />
        <button>
          Search
        </button>
        <p>
          lorem
        </p>
      </form>
    </div>
    <div>
      <ul>
        <li>
          Escreva o nome do seu pokemon
        </li>
      </ul>
    </div>
  </div>
</div>
`;
